code/modules/recycling/conveyor2.dm

```diff
@@ -44,107 +89,117 @@
+	if(machine_stat & (BROKEN | NOPOWER))
 		return
-	if(!operating)
+
+	//If the conveyor is broken or already moving items
+	if(!operating || conveying)
 		return
 
-	affecting = loc.contents - src		// moved items will be all in loc
-	INVOKE_NEXT_TICK(src, .proc/propogate)	// slight delay to prevent infinite propagation due to map order
-
-/obj/machinery/conveyor/proc/propogate()
-	var/items_moved = 0
-	for(var/atom/movable/A in affecting)
-		if(!A.anchored)
-			if(A.loc == src.loc) // prevents the object from being affected if it's not currently here.
-				step(A,movedir)
-				items_moved++
-		if(items_moved >= 10)
-			break
+	//get the first 30 items in contents
+	var/turf/locturf = loc
+	var/list/items = locturf.contents - src - locturf.lighting_object
+	if(!LAZYLEN(items))//Dont do anything at all if theres nothing there but the conveyor
+		return
+	var/list/affecting
+	if(length(items) > MAX_CONVEYOR_ITEMS_MOVE)
+		affecting = items.Copy(1, MAX_CONVEYOR_ITEMS_MOVE + 1)//Lists start at 1 lol
+	else
+		affecting = items
+	conveying = TRUE
+
+	INVOKE_NEXT_TICK(src, .proc/convey, affecting)//Movement effect
+
+/obj/machinery/conveyor/proc/convey(list/affecting)
+	for(var/am in affecting)
+		if(!ismovable(am))	//This is like a third faster than for(var/atom/movable in affecting)
+			continue
+		var/atom/movable/movable_thing = am
+		//Give this a chance to yield if the server is busy
+		stoplag()
+		if(QDELETED(movable_thing) || (movable_thing.loc != loc))
+			continue
+		if(iseffect(movable_thing) || isdead(movable_thing))
+			continue
+		if(!movable_thing.anchored)
+			step(movable_thing, movedir)
+	conveying = FALSE
 
 // attack with item, place item on conveyor
 /obj/machinery/conveyor/attackby(obj/item/I, mob/user, params)
-	. = ..()
-
-	if(istype(I, /obj/item/grab))
-		var/obj/item/grab/G = I
-
-		if(ismob(G.grabbed_thing))
-			var/mob/GM = G.grabbed_thing
-			step(GM, get_dir(GM, src))
-
+	if(I.tool_behaviour == TOOL_CROWBAR)
+		user.visible_message("<span class='notice'>[user] struggles to pry up \the [src] with \the [I].</span>", \
+		"<span class='notice'>You struggle to pry up \the [src] with \the [I].</span>")
+		if(I.use_tool(src, user, 40, volume=40))
+			if(!(machine_stat & BROKEN))
+				new /obj/item/stack/conveyor(loc, 1, TRUE, id)
+			to_chat(user, "<span class='notice'>You remove the conveyor belt.</span>")
+			qdel(src)
+
+	else if(I.tool_behaviour == TOOL_WRENCH)
+		if(!(machine_stat & BROKEN))
+			I.play_tool_sound(src)
+			setDir(turn(dir,-45))
+			update_move_direction()
+			to_chat(user, "<span class='notice'>You rotate [src].</span>")
+
+	else if(I.tool_behaviour == TOOL_SCREWDRIVER)
+		if(!(machine_stat & BROKEN))
+			verted = verted * -1
+			update_move_direction()
+			to_chat(user, "<span class='notice'>You reverse [src]'s direction.</span>")
+
+	else if(user.a_intent != INTENT_HARM)
+		user.transferItemToLoc(I, drop_location())
 	else
-		user.transferItemToLoc(I, loc)
+		return ..()
```

Comment:
Ain't these supposed to be `screwdriver_act()` and what not?

Reply:
crap